### PR TITLE
Update docs for `--light` mode

### PR DESCRIPTION
**Pull request by piotr-iohk**
- [x] I have updated docs (cli.md and use case) for upcoming `--light` mode

### Issue Number

ADP-1426

### CODE REVIEW

File: docs/user-guide/cli.md [hunk 1]
```diff
@@ -111,18 +111,24 @@ Serve API that listens for commands/actions. Before launching user should start
...
-                           '*'. (default: 127.0.0.1)
+  --listen-address HOST    Specification of which host to bind the API server
+                           to. Can be an IPv[46] address, hostname, or '*'.
+                           (default: 127.0.0.1)
+  --node-socket FILE       Path to the node's domain socket file (POSIX) or pipe
+                           name (Windows). Note: Maximum length for POSIX socket
+                           files is approx. 100 bytes. Note: Windows named pipes
+                           are of the form \\.\pipe\cardano-node
+  --light                  Enable light mode
+  --blockfrost-token-file FILE
```
Comment (Unisay) [inline docs/user-guide/cli.md:136]: These options are mutually exclusive. I don't know if this documentation is supposed to indicate it somehow.

File: docs/user-guide/common-use-cases/how-to-start-wallet-server.md [hunk 2]
```diff
@@ -44,6 +50,40 @@ $ curl -X GET http://localhost:8090/v2/network/information | jq .sync_progress
 }
 ```
 
-## Lightwallet mode
+## Light mode
+
+> :warning: This mode is currently **under development**. Please note that some parts may be available only in `master` branch.
+
+You can start your cardano-wallet server also in **light mode**. As opposed to full-node mode, in light your wallet is not connected to locally running instance of `cardano-node`. Instead it relays on external source of blockchain data. This significantly improves synchronization speed of the wallet and also you don't need to spend time for synchronizing the node itself. The downside is that external source of data is obviously less trusted than the local one provided by your own `cardano-node` instance.
```
Comment (Unisay) [inline docs/user-guide/common-use-cases/how-to-start-wallet-server.md]:
```suggestion
You can start your cardano-wallet server also in **light mode**. As opposed to full-node mode, in light-mode your wallet is not connected to a locally running instance of `cardano-node`. Instead it relays on external source of blockchain data. This significantly improves synchronization speed of the wallet and also removes the need to spend time synchronizing the node itself. The downside is that external source of data is obviously less trusted than the local one provided by your own `cardano-node` instance.
```

**Review by Unisay — APPROVED** [review]

File: docs/user-guide/cli.md (hunk 1 above)
Comment (piotr-iohk) [reply docs/user-guide/cli.md:136]: It's a `copy-paste` of  `Available options` from `cardano-wallet serve --help`... the mutual exclusiveness is there indicated by `usage`:
```
Usage: cardano-wallet serve [--listen-address HOST] 
                            (--node-socket FILE | --light
                              --blockfrost-token-file FILE) 
                            [--random-port | --port INT] 
                            [--tls-ca-cert FILE --tls-sv-cert FILE
                              --tls-sv-key FILE] 
                            (--mainnet | --testnet FILE | --staging FILE) 
                            [--database DIR] [--sync-tolerance DURATION] 
                            [--shutdown-handler] 
                            [--pool-metadata-fetching ( none | direct | SMASH-URL )]
                            [--token-metadata-server URL] 
                            [--trace-NAME SEVERITY]
```
Perhaps a note in docs would be handy too.

**Pull request merged**